[REVIEW] Default NVLink cluster configuration and updated README (#11)
rapidsai/tpcx-bb: README.md

```bash
conda list | grep xbb
xbb-tools                 0.2                      pypi_0    pypi
```

Note that this Conda environment needs to be replicated or installed manually on all nodes, which will allow starting one dask-cuda-worker per node.

## Cluster Setup

We use the `dask-scheduler` and `dask-cuda-worker` command line interfaces to start a Dask cluster. We provide a `cluster_configuration` directory with a bash script to help you set up an NVLink-enabled cluster using UCX.

> Before running the script, you'll need to make a few small changes specific to your machines.

Reviewer: ```suggestion
Before running the script, you'll make a changes specific to your environment.
```

Author: nitpicking your nitpitck :)

To start up the cluster, please run the following on every node.

```bash
bash cluster-startup.sh NVLINK
```

## Running the Queries

To run a query, starting from the repository root, go to the `queries` directory:

```bash
cd tpcx_bb/queries/
```

Choose a query to run, and `cd` to that directory. We'll pick query 07.

```bash
cd q07
```

Activate the conda environment you've created.

The queries assume that they can attach to a running Dask cluster. Command line arguments are used to determine the cluster and dataset on which to run the queries. The following is an example of running query 07.

```bash
python tpcx_bb_query_07.py --data_dir=$DATA_DIR --cluster_host=$SCHEDULER_IP --output_dir=$OUTPUT_DIR
```

- `data_dir` points to the location of the data
- `cluster_host` corresponds to the schedyler address of the running Dask cluster
    - In this case, this query would attempt to connect to a cluster running at `$SCHEDULER_IP`, which we configured during cluster startup
- `output_dir` points to where the queries should write output files. If not specified, queries will write output into their respective directories

## Performance Tracking

> This repository contains support for performance-tracking automation using Google Sheets. If you install `gspread` and `oauth2client` in your conda environment, you will be able to pass `--sheet` and `--tab` arguments the query CLI to push performance metrics to Google Sheets. If you do not have these libraries installed, you will see `Please install gspread and oauth2client to use Google Sheets automation` after running a query. It's fine to ignore this output.

Reviewer: ```suggestion
`gspread` and `oauth2client` are included in the provided conda environment spec. Passing `--sheet` and `--tab` arguments during query execution will log query runtime in a Google Sheet. If you do not have these libraries installed, you will see `Please install gspread and oauth2client to use Google Sheets automation` after running a query. It's fine to ignore this output.
```

Reviewer: Just realized these aren't actually in the conda environment file yet. Please add them.

Author: This is not currently true, but we can add them

Author: We’ll need to update the utilities code again if we add gspread as a dependency. Right now, the code will fail gracefully if gspread is not available. but, if it is, we will fail loudly if the google sheets configuration is incorrect.

It may be worth doing this in a follow-up

Author: Marking this as for a follow-up. Updated this section to address some of the edits.

Expanding on the example above, you could do the following:

```bash
python tpcx_bb_query_07.py --data_dir=$DATA_DIR --cluster_host=$SCHEDULER_IP --output_dir=$OUTPUT_DIR --sheet=TPCx-BB --tab="SF1000 Benchmarking Matrix"
```

### Running all of the Queries

> As a convenience, you can sequentially run all the queries with the provided `benchmark_runner.py` script. Instead of passing the command line arguments directly, you will need to fill in the same information in `benchmark_config.yaml`, which can be found in the `benchmark_runner/` directory.

Reviewer: ```suggestion
The included `benchmark_runner.py` script will run all queries sequentially. Configuration for this type of end to end run is specified in `benchmark_runner/benchmark_config.yaml`.
```

> To execute this, run the following from `tpcx_bb/`:

Reviewer: ```suggestion
To run all queries, cd to `tpcx_bb/` and:
```
